crates/query-engine/sql/src/sql/rewrites/constant_folding.rs

```diff
@@ -94,6 +94,7 @@ pub fn normalize_cte(mut cte: CommonTableExpression) -> CommonTableExpression {
                 })
                 .collect(),
         ),
+        CTExpr::Delete(delete) => CTExpr::Delete(delete), // TODO: normalize this properly
```

Comment:
This should be easy - normalize the where and mutate the delete. right?

Reply:
Oh ofc! Forgot this. Indeed, nothing too complicated I'd imagine.

Reply:
Done!